src/TicketController.java

```diff
@@ -0,0 +1,63 @@
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.util.Date;
+import java.util.Scanner;
+
+public class TicketController {
+    public static void main(String[] args) throws ParseException {
+        TicketMachine ticketMachine = new TicketMachine();
+
+        Passenger[] passengers = new Passenger[3];
+        passengers[0] = new Passenger("Wojciech", "Kostecki");
+        passengers[1] = new Passenger("Rafał", "Bobko");
+        passengers[2] = new Passenger("Damian", "Podbilski");
+
+        ticketMachine.buyTicket(passengers[0], 30);
+        ticketMachine.buyTicket(passengers[1], 20);
+        ticketMachine.buyTicket(passengers[2], 15);
+
+        printPassengerWithoutTicket(validateTickets(passengers));
+    }
+
+    private static Passenger[] validateTickets(Passenger[] passengers) throws ParseException {
+        Scanner scanner = new Scanner(System.in);
+        System.out.println("Data i godzina kontroli w formacie: yyyy-MM-dd HH:mm");
+        String date = scanner.nextLine();
+        SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd HH:mm");
+        Date controlDate = sdf.parse(date);
+
+        int passengersWithoutTicket = 0;
+
+        for (Passenger passenger : passengers) {
+            LocalDateTime ticketDate = passenger.getTicket().getBuyingTime().
+                    plusMinutes(passenger.getTicket().getValidityTime());
+            Date ticketExpiryDate = Date.from(ticketDate.atZone(ZoneId.systemDefault()).toInstant());
+            if (controlDate.after(ticketExpiryDate)) {
+                passengersWithoutTicket++;
+            }
+        }
+
+        Passenger[] withoutTicket = new Passenger[passengersWithoutTicket];
+
+        for (int i = 0, j = 0; i < passengers.length; i++) {
+            LocalDateTime ticketDate = passengers[i].getTicket().getBuyingTime().
+                    plusMinutes(passengers[i].getTicket().getValidityTime());
+            Date ticketExpiryDate = Date.from(ticketDate.atZone(ZoneId.systemDefault()).toInstant());
```

Comment:
Try to stick to one date type. The best one for now in Java is LocalDate/LocalDateTime and you should use this one.